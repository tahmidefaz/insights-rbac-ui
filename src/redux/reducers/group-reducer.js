import {
  FETCH_GROUP,
  FETCH_GROUPS,
  RESET_SELECTED_GROUP
} from '../../redux/action-types';

// Initial State
export const groupsInitialState = {
  groups: {
    data: [],
    meta: {
      count: 0,
      limit: 10,
      offset: 0
    }
  },
  selectedGroup: {},
  isLoading: false
};

const setLoadingState = state => ({ ...state, isLoading: true });
const setGroups = (state, { payload }) => ({ ...state, groups: payload, isLoading: false });
const selectGroup = (state, { payload }) => ({ ...state, selectedGroup: payload, isLoading: false });
const resetSelectedGroup = state => ({ ...state, selectedGroup: undefined });

export default {
  [`${FETCH_GROUPS}_PENDING`]: setLoadingState,
  [`${FETCH_GROUPS}_FULFILLED`]: setGroups,
  [`${FETCH_GROUP}_PENDING`]: setLoadingState,
  [`${FETCH_GROUP}_FULFILLED`]: selectGroup,
  [RESET_SELECTED_GROUP]: resetSelectedGroup
};
