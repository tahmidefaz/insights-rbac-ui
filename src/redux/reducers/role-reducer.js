import {
  FETCH_ROLE,
  FETCH_ROLES
} from '../../redux/action-types';

// Initial State
export const rolesInitialState = {
  isLoading: false,
  isRecordLoading: false,
  roles: {
    data: [],
    meta: {
      count: 0,
      limit: 10,
      offset: 0
    }
  },
  selectedRole: {}
};

const setLoadingState = state => ({ ...state, isLoading: true });
const setRecordLoadingState = state => ({ ...state, isRecordLoading: true });
const setRole = (state, { payload }) => ({ ...state, selectedRole: payload, isRecordLoading: false });
const setRoles = (state, { payload }) => ({ ...state, roles: payload, isLoading: false });

export default {
  [`${FETCH_ROLE}_FULFILLED`]: setRole,
  [`${FETCH_ROLE}_PENDING`]: setRecordLoadingState,
  [`${FETCH_ROLES}_FULFILLED`]: setRoles,
  [`${FETCH_ROLES}_PENDING`]: setLoadingState
};
