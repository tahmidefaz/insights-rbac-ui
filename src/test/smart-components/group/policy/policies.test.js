import React from 'react';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store' ;
import { shallowToJson } from 'enzyme-to-json';
import promiseMiddleware from 'redux-promise-middleware';
import GroupPolicies from '../../../../smart-components/group/policy/policies';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications';
import { policiesInitialState } from '../../../../redux/reducers/policy-reducer';

describe('<GroupPolicies />', () => {

  let initialProps;
  const middlewares = [ thunk, promiseMiddleware(), notificationsMiddleware() ];
  let mockStore;
  let initialState;

  beforeEach(() => {
    initialProps = {};
    mockStore = configureStore(middlewares);
    initialState = { policyReducer: { ...policiesInitialState, isLoading: false }};
  });

  it('should render correctly', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<GroupPolicies store={ store } { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly in loading state', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<GroupPolicies store={ store } { ...initialProps } isLoading />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
