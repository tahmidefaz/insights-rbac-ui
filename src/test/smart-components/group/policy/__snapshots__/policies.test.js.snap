// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupPolicies /> should render correctly 1`] = `
<GroupPolicies
  fetchGroupPolicies={[Function]}
  isLoading={false}
  pagination={
    Object {
      "count": 0,
      "limit": 10,
      "offset": 0,
    }
  }
  policies={Array []}
  removePolicy={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`<GroupPolicies /> should render correctly in loading state 1`] = `
<GroupPolicies
  fetchGroupPolicies={[Function]}
  isLoading={false}
  pagination={
    Object {
      "count": 0,
      "limit": 10,
      "offset": 0,
    }
  }
  policies={Array []}
  removePolicy={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
